docs/eBpfStringsAPI.md

```diff
@@ -0,0 +1,91 @@
+# eBPF Strings Library API proposal
+
+## Overview
+
+A majority of use cases and examples of eBPF functions are focused on operating on fixed fields in buffers; network structures with known sizes and offsets, bit-fields, and the like. The programs we can write with the available tooling are powerful, but fall short of the needs of Application-layer customers like HTTP servers, where string operations are mandatory to interact with HTTP requests and responses.
+
+This has precedent; in [`ebpf_core_helper_function_prototype_array[]`](../libs/execution_context/ebpf_general_helpers.c) there are `bpf_memcmp()`, `bpf_memcpy()`, `bpf_memmove()`, and `bpf_memset()` -- all of which are also derived from `<string.h>` functions.
+
+## Scenarios
+
+### Finding a token in a header
+
+Assuming an appropriate extension existed to interact with an in-flight HTTP request, a customer wants to search through the Accept-Encoding header, because some older application can't handle gzip appropriately. (I'm sure this is a real use case somewhere.) Error handling is elided in this example for brevity.
```

Comment:
```suggestion
Assuming an appropriate extension existed to interact with an in-flight HTTP request, a developer wants to search through the Accept-Encoding header, because some older application can't handle gzip appropriately. (I'm sure this is a real use case somewhere.) Error handling is elided in this example for brevity.
```